### PR TITLE
Document the new dialog API

### PR DESCRIPTION
**Pull request by dtdesign**
n/t

### CODE REVIEW

File: docs/javascript/components_confirmation.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The confirmation question should hint the severity of the action, in particular whether or not it is destructive.
+Destructive actions are those that cannot be undone and either cause a permanent mutation or that cause data loss.
+All questions should be phrased in one or two ways depending on the action.
+
+Destructive action:
+> Are you sure you want to […]?
+
+All other actions:
+> Do you want to […]
```
Comment (TimWolla) [inline docs/javascript/components_confirmation.md]: Should this also mention the German variant?

File: docs/javascript/components_confirmation.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+All three presets require the title of the related object as part of the question asked to the user.
+
+### Soft Delete
+
+Soft deleting objects with an optional input field for a reason:
+
+```ts
+const askForReason = true;
+const { result, reason } = await confirmationFactory()
+  .softDelete(theObjectName, askForReason);
```
Comment (TimWolla) [inline docs/javascript/components_confirmation.md]: What will the return value look like, if no reason is requested? Will `reason` just be missing?

File: docs/javascript/components_dialog.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
+# Dialogs - JavaScript API
+
+Modal dialogs are a powerful tool to draw the viewer‘s attention to an important message, question or form.
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md]:
```suggestion
Modal dialogs are a powerful tool to draw the viewer’s attention to an important message, question or form.
```

File: docs/javascript/components_dialog.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+```ts
+const dialog = dialogFactory()
+  .fromHtml("<p>Hello World</p>")
+  .withoutControls();
+dialog.show("Greetings from my dialog");
+```
+
+### When to Use
+
+The short answer is: Don‘t.
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md]:
```suggestion
The short answer is: Don’t.
```

File: docs/javascript/components_dialog.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
+# Dialogs - JavaScript API
+
+Modal dialogs are a powerful tool to draw the viewer‘s attention to an important message, question or form.
+Dialogs naturally interrupt the workflow and prevent the navigation to other sections by making other elements on the page inert.
+
+WoltLab Suite 6.0 ships with four different types of dialogs.
+
+## Dialogs Without Controls
+
+Dialogs may contain just an explanation or extra information that should be presented to the viewer without requiring any further interaction.
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md:20]: What does “without controls” imply? Will the close button be available?

File: docs/javascript/components_dialog.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+## Confirmation
+
+Confirmation dialogs are supported through a separate factory function that provides a set of presets as well as a generic API. Please see the separate documentation for [confirmation dialogs](components_confirmation.md) to learn more.
+
+## Prompts
+
+The most common type of dialogs are prompts that are similar to confirmation dialogs, but without the restrictions and with a regular title.
+These dialogs can be used universally and provide a submit and cancel button by default.
+In addition they offer an “extra” button that is placed to the left of the default buttons are can be used to offer a single additional action.
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md:88]: When would this be useful? Add an example.

File: docs/javascript/components_dialog.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+    event.preventDefault();
+
+    // Display an inline error message.
+  }
+});
+```
+
+## Code Examples
+
+TODO
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md]: TODO

**Review by TimWolla — COMMENTED** [review]
Overall this reads very well.

File: docs/javascript/components_confirmation.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+const result = await confirmationFactory()
+  .custom("Do you want a cookie?")
+  .withoutMessage();
+if (result) {
+  // User has confirmed the dialog.
+}
+```
+
+Confirmation dialogs are a special type that use the `role="alertdialog"` attribute and will always include a cancel button.
+The dialog itself will be limited to a width of 500px, the title can wrap into multiple lines and there will be no „X“ button to close the dialog.
```
Comment (TimWolla) [inline docs/javascript/components_confirmation.md]:
```suggestion
The dialog itself will be limited to a width of 500px, the title can wrap into multiple lines and there will be no “X” button to close the dialog.
```

File: docs/javascript/components_confirmation.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+## Proper Wording
+
+The confirmation question should hint the severity of the action, in particular whether or not it is destructive.
+Destructive actions are those that cannot be undone and either cause a permanent mutation or that cause data loss.
+All questions should be phrased in one or two ways depending on the action.
+
+Destructive action:
+
+> Are you sure you want to delete “Example Object”?
+> (German) Wollen Sie „Beispiel-Objekt” wirklich löschen?
```
Comment (TimWolla) [inline docs/javascript/components_confirmation.md]:
```suggestion
> (German) Wollen Sie „Beispiel-Objekt“ wirklich löschen?
```
German closing quote is identical to the English opening quote.

English opening: Thick at the bottom, thin at the top.
English closing: Thick at the top, thin at the bottom.
German opening: Thick at the top, thin at the bottom.
German closing: Thick at the bottom, thin at the top.

Thus: Opening and closing are identical under point reflection with a 180° rotation.

File: docs/javascript/components_confirmation.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+All other actions:
+
+> Do you want to move “Example Object” to the trash bin?
+> (German) Möchten Sie „Beispiel-Objekt“ in den Papierkorb verschieben?
```
Comment (TimWolla) [inline docs/javascript/components_confirmation.md:41]: It’s correct for this one :smiley:

File: docs/javascript/components_dialog.md [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+
+If possible these dialogs should be avoided and the content is presented in a more suitable way, for example, as a flyout or by showing content on an existing or new page.
+
+## Alerts
+
+Alerts are designed to inform the user of something important that requires no further action by the user.
+Typical examples for alerts are error messages or warnings.
+
+An alert will only provide a single button to acknowledge the dialog and must not contain interactive content.
+The dialog itself will be limited to a width of 500px, the title can wrap into multiple lines and there will be no „X“ button to close the dialog.
```
Comment (TimWolla) [inline docs/javascript/components_dialog.md]:
```suggestion
The dialog itself will be limited to a width of 500px, the title can wrap into multiple lines and there will be no “X” button to close the dialog.
```

**Review by TimWolla — APPROVED** [review]
Apart from the quote characters suggestions this LGTM now.

File: docs/javascript/components_confirmation.md (hunk 9 above)
Comment (dtdesign) [reply docs/javascript/components_confirmation.md]: Thanks, but I’m well aware of this. Mixing both languages sometimes cause me to unconsciously use the wrong keystrokes for the quote marks.
* Option+^ // Shift+Option+^ = „…“
* Option+2 // Shift+Option+2 = “…”

**Comment by dtdesign** [comment]
Depends on WoltLab/WCF#5014

**Pull request merged**